catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/PhiroColorBrickTest.java

```diff
@@ -0,0 +1,194 @@
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.app;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.PhiroRGBLightBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.actions.CustomActions;
+import org.catrobat.catroid.uiespresso.util.matchers.ScriptListMatchers;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.hamcrest.Matchers.instanceOf;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroColorBrickTest {
+	private static final int SET_RED = 50;
+	private static final int SET_GREEN = 100;
+	private static final int SET_BLUE = 200;
+	private static final int INIT_COLOR = 0;
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		Script script = BrickTestUtils.createProjectAndGetStartScript("PhiroColorBrickTest");
+		script.addBrick(new PhiroRGBLightBrick(PhiroRGBLightBrick.Eye.BOTH, INIT_COLOR, INIT_COLOR, INIT_COLOR));
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testPhiroLightSpinner() {
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_phiro_rgb_led_action);
+
+		List<Integer> spinnerValues = new ArrayList<>();
+		spinnerValues.add(R.string.phiro_light_both);
+		spinnerValues.add(R.string.phiro_light_left);
+		spinnerValues.add(R.string.phiro_light_right);
+		onBrickAtPosition(1).onSpinner(R.id.brick_phiro_rgb_light_spinner).checkValuesAvailable(spinnerValues);
+
+		onData(instanceOf(PhiroRGBLightBrick.class))
```

Comment:
please use the brickDataInteractionWrapper....

and if youre clicking through all the available spinner values, dont, you already asserted above that they are available.